### PR TITLE
Fix `Box<T>` and `Nullable<T>` as entry point arguments.

### PR DESCRIPTION
**Pull request by azteca1998**
Something I missed from the previous fix.

## Checklist
- [ ] Linked to Github Issue
- [ ] Unit tests added
- [x] Integration tests added.
- [ ] This change requires new documentation.
  - [ ] Documentation has been added/updated.

### CODE REVIEW

**Review by edg-l — APPROVED** [review]

File: src/arch.rs [hunk 1]
```diff
@@ -57,14 +57,49 @@ impl<'a> JitValueWithInfoWrapper<'a> {
...
-                CoreTypeConcrete::Box(info)
-                | CoreTypeConcrete::NonZero(info)
-                | CoreTypeConcrete::Nullable(info)
-                | CoreTypeConcrete::Snapshot(info),
-            ) => {
-                // TODO: Allocate and use to_jit().
+            (value, CoreTypeConcrete::Box(info)) => {
+                let ptr = value
+                    .to_jit(self.arena, self.registry, self.type_id)
+                    .unwrap();
```
Comment (pefontana) [inline src/arch.rs:63]: Are these unwraps safe?
Should be implement an error handling here? Maybe we can make it in another PR if it is necesarry

File: src/arch.rs [hunk 2]
```diff
@@ -57,14 +57,49 @@ impl<'a> JitValueWithInfoWrapper<'a> {
...
+
+                heap_ptr.to_bytes(buffer);
+            }
+            (value, CoreTypeConcrete::Nullable(info)) => {
+                if matches!(value, JitValue::Null) {
+                    null::<()>().to_bytes(buffer);
+                } else {
+                    let ptr = value
+                        .to_jit(self.arena, self.registry, self.type_id)
+                        .unwrap();
```
Comment (pefontana) [inline src/arch.rs:85]: same here

**Review by pefontana — CHANGES_REQUESTED** [review]

File: src/arch.rs (hunk 1 above)
Comment (azteca1998) [reply src/arch.rs:63]: Ideally, yes. It could technically fail, but it'll most likely be because of an invalid Sierra program or internal Cairo error (the libraries we're using).
We can turn it into an error later on along with all the others once we fix our current bugs.

File: src/arch.rs (hunk 2 above)
Comment (pefontana) [reply src/arch.rs:85]: #748

**Review by pefontana — APPROVED** [review]
